rosbag2_compression/test/rosbag2_compression/test_zstd_compressor.cpp

```diff
@@ -0,0 +1,60 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <fstream>
+#include <string>
+#include <rclcpp/rclcpp.hpp>
+
+#include "rosbag2_storage/filesystem_helper.hpp"
+#include "rosbag2_test_common/temporary_directory_fixture.hpp"
+
+#include "gmock/gmock.h"
+
+#include "rosbag2_compression/zstd_compressor.hpp"
+
+class FilesystemHelperFixture : public rosbag2_test_common::TemporaryDirectoryFixture {};
+
+/**
+ * Creates a 27 MiB file.
+ * \param uri File path to write file.
+ * \param num_iterations Number of times to write garbage statement.
+ */
+void create_garbage_file(const std::string & uri, int num_iterations=984467)
+{
+  std::ofstream out(uri);
+  for (int i = 0; i < num_iterations; i++) {
+    out << "Garbage";
+  }
+  out.close();
```

Comment:
`std:ofstream::close` is not needed since it will close automatically when it leaves scope.